### PR TITLE
Ermshark fixes

### PR DESCRIPTION
**Pull request by Govorunb**
This should fix at least a few issues with ermsharks multiplying out of control.

Some documentation here: Govorunb/Vedal-Subnautica-Mod#1

I didn't do anything to fix the "baby shark turns to full shark" issue but after these changes I couldn't reproduce it anymore so shrug

### CODE REVIEW

**Comment by budwheizzah** [comment]
I tested the filtered biome spawning included in this fix. I concur, it seems to improve creature distribution. Higher probabilities seem to be necessary because this selects far fewer places to spawn the creatures in. Well done. I applied it to my Tutel WIP as well.

File: SCHIZO/Creatures/Ermshark/Ermshark.cs [hunk 1]
```diff
@@ -12,6 +12,12 @@ public void OnTakeDamage(DamageInfo damageInfo)
     {
         if (liveMixin.health > 0) return;
 
+        if (damageInfo.dealer != Player.main.gameObject)
```
Comment (Alexejhero) [inline SCHIZO/Creatures/Ermshark/Ermshark.cs]: Does this work for all damage types a player can cause? Propulsion Cannon, repulsion cannon, knife, thermoblade, ramming with seamoth, etc?

File: SCHIZO/Creatures/Ermfish/ErmfishLoader.cs [hunk 2]
```diff
@@ -77,7 +77,8 @@ private static void LoadErmfish()
         });
 
         List<LootDistributionData.BiomeData> biomes = new();
-		foreach (BiomeType biome in Enum.GetValues(typeof(BiomeType)).Cast<BiomeType>())
+		foreach (BiomeType biome in Enum.GetValues(typeof(BiomeType)).Cast<BiomeType>()
+            .Where(biome => biome.ToString().EndsWith("CreatureOnly")))
```
Comment (Alexejhero) [inline SCHIZO/Creatures/Ermfish/ErmfishLoader.cs:81]: What's going on here?

File: SCHIZO/Creatures/Ermshark/ErmsharkPatches.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                CodeInstruction loadDealerInst = codes[i + 1];
+                CodeInstruction takeDamageCall = codes[i + 2];
+                if (loadDealerInst.opcode == OpCodes.Ldnull && takeDamageCall.Calls(takeDamageMethod))
+                {
+                    //LOGGER.LogDebug($"Before patch:\n\t{string.Join("\n\t", codes.GetRange(i, 3))}");
+                    loadDealerInst.opcode = OpCodes.Ldsfld;
+                    loadDealerInst.operand = mainPlayerField;
+                    CodeInstruction loadPlayerObject = new(OpCodes.Callvirt, gameObjectGetter);
+                    codes.Insert(i+2, loadPlayerObject);
```
Comment (Alexejhero) [inline SCHIZO/Creatures/Ermshark/ErmsharkPatches.cs]: You aren't moving any labels, does this break other code paths?

(Sorry I can't check stuff ATM I'm just guessing what problems there might be)

File: SCHIZO/Creatures/Ermshark/Ermshark.cs (hunk 1 above)
Comment (Govorunb) [reply SCHIZO/Creatures/Ermshark/Ermshark.cs]: Only the knives set the damage dealer - I think it's funnier if you can only kill the shark by stabbing but I can probably get vehicle collisions working too (lmk)

File: SCHIZO/Creatures/Ermfish/ErmfishLoader.cs (hunk 2 above)
Comment (Govorunb) [reply SCHIZO/Creatures/Ermfish/ErmfishLoader.cs:81]: There are 495 enum values in all and most of them don't actually represent a good spawn location for any kind of creature - wrecks, walls, floors, etc. Most common fish have about 20-40 biomes assigned to them, and this change brings the previous 495 down to ~50. I've been playing this branch on my save over the weekend and haven't noticed too big of a change in spawn rates - ermfish and ermsharks both still show up everywhere

File: SCHIZO/Creatures/Ermshark/ErmsharkPatches.cs (hunk 3 above)
Comment (Govorunb) [reply SCHIZO/Creatures/Ermshark/ErmsharkPatches.cs]: I looked it up while I was making the patch and some docs somewhere said Harmony manages the labels automatically. At the very least, I haven't encountered any issues playing the game

File: SCHIZO/Creatures/Ermfish/ErmfishLoader.cs (hunk 2 above)
Comment (budwheizzah) [reply SCHIZO/Creatures/Ermfish/ErmfishLoader.cs:81]: I just messed around with spawn biomes tonight and the ones Gov is picking are perfect for fish. They're basically always two different levels of open water (deep or shallow). You're guaranteed not to have accidental in-terrain spawns, and I believe those zones are specifically designated for fish to spawn. Might have to increase/tweak probabilities, but this is likely better than trying to spawn in all biomes.

**Pull request merged**